src/producer_processor_saver/LogicMultiple.java

```diff
@@ -0,0 +1,222 @@
+package producer_processor_saver;
+
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class LogicMultiple {
+
+    /**
+     * Queue to emulate loading from disk.
+     */
+    private Queue<Integer> loadingQueue = new LinkedList<>();
+
+    /**
+     * Queue to emulate processed data ready for storing.
+     */
+    private Queue<String> processedQueue = new LinkedList<>();
+
+    private int maxCount;
+    private int maxLoadStorage;
+    private int maxProcessedStorage;
+    private int loadTime;
+    private int processingTime;
+    private int storeTime;
+
+    private Lock loadedResourcesLock = new ReentrantLock();
+    private Lock processedDataLock = new ReentrantLock();
+    private Object maxCountLoadingLock = new Object();
+    private Object maxCountProcessingLock = new Object();
+    private Object maxCountStoreLock = new Object();
+
+    private Condition nonEmptyLoadCondition = loadedResourcesLock.newCondition();
+    private Condition notFullLoadCondition = loadedResourcesLock.newCondition();
+    private Condition nonEmptyProcessedCondtion = processedDataLock.newCondition();
+    private Condition notFullProcessedCodntion = processedDataLock.newCondition();
+
+
+    public LogicMultiple(int maxCount, int maxLoadStorage, int maxProcessedStorage, int loadTime, int processingTime,
+                 int storeTime) {
+        this.maxCount = maxCount;
+        this.maxLoadStorage = maxLoadStorage;
+        this.maxProcessedStorage = maxProcessedStorage;
+        this.loadTime = loadTime;
+        this.processingTime = processingTime;
+        this.storeTime = storeTime;
+    }
+
+    private Integer loadNext() {
+        synchronized (maxCountLoadingLock) {
+            if (counterLoading >= maxCount) {
+                return null;
+            } else {
+                counterLoading++;
+            }
+        }
+        return load(counterLoading - 1);
```

Comment:
You're reading from unprotected variable. The value here can be anything and multiple threads may return same number. The easiest but way from ideal solution is to move the return into the else branch above.